csharp/ql/lib/semmle/code/csharp/Type.qll

```diff
@@ -449,6 +451,14 @@ class SimpleType extends ValueType, @simple_type {
   override SystemNamespace getDeclaringNamespace() { any() }
 }
 
+/**
+ * A `record` like type.
+ * This can be either a `class` or a `struct`.
+ */
+abstract class RecordType extends ValueOrRefType {
```

Comment:
Remove `abstract`.

Reply:
May I ask why?
When declared abstract we are forcing RecordType = RecordClass \union RecordStruct. ValueOrRefType is pretty general and maybe "one day" other types will get the "record" modifier. Do we then automatically wan't them to be a part of RecordType (this might then break FlowSummary.qll implementation)? Right now there is a built in assumption that the RecordType is the disjoint union of two other sets.

Reply:
Given the name `RecordType`, I would expect it to include all `record` types (suppose `record interface` became a thing). Also, we never use `abstract` classes in user-facing libraries for closed unions, as a user writing `extends` would than mistakenly be adding to the type, rather than refining it.

Reply:
Ah yes, you are absolutely right - exposing abstract towards clients is not a good idea! I didn't think about that! Thank you for the clarification!